android/src/main/java/dev/robingenz/capacitorjs/plugins/firebase/auth/handlers/AppleAuthProviderHandler.java

```diff
@@ -0,0 +1,126 @@
+package dev.robingenz.capacitorjs.plugins.firebase.auth.handlers;
+
+import android.util.Log;
+import com.getcapacitor.JSArray;
+import com.getcapacitor.JSObject;
+import com.getcapacitor.PluginCall;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.OAuthCredential;
+import com.google.firebase.auth.OAuthProvider;
+import dev.robingenz.capacitorjs.plugins.firebase.auth.FirebaseAuthentication;
+import dev.robingenz.capacitorjs.plugins.firebase.auth.FirebaseAuthenticationPlugin;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CodingErrorAction;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.List;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class AppleAuthProviderHandler {
+
+    private final FirebaseAuthentication pluginImplementation;
+    private String nonce;
+
+    public AppleAuthProviderHandler(FirebaseAuthentication pluginImplementation) {
+        this.pluginImplementation = pluginImplementation;
+    }
+
+    // From https://firebase.google.com/docs/auth/android/apple#advanced_handle_the_sign-in_flow_manually
+    private String generateNonce(int length) {
+        SecureRandom generator = new SecureRandom();
+
+        CharsetDecoder charsetDecoder = StandardCharsets.US_ASCII.newDecoder();
+        charsetDecoder.onUnmappableCharacter(CodingErrorAction.IGNORE);
+        charsetDecoder.onMalformedInput(CodingErrorAction.IGNORE);
+
+        byte[] bytes = new byte[length];
+        ByteBuffer inBuffer = ByteBuffer.wrap(bytes);
+        CharBuffer outBuffer = CharBuffer.allocate(length);
+        while (outBuffer.hasRemaining()) {
+            generator.nextBytes(bytes);
+            inBuffer.rewind();
+            charsetDecoder.reset();
+            charsetDecoder.decode(inBuffer, outBuffer, false);
+        }
+        outBuffer.flip();
+        return outBuffer.toString();
+    }
+
+    private String sha256(String s) throws NoSuchAlgorithmException {
+        MessageDigest md = MessageDigest.getInstance("SHA-256");
+        byte[] digest = md.digest(s.getBytes());
+        StringBuilder hash = new StringBuilder();
+        for (byte c : digest) {
+            hash.append(String.format("%02x", c));
+        }
+        return hash.toString();
+    }
+
+    public void signIn(PluginCall call) {
+        OAuthProvider.Builder provider = OAuthProvider.newBuilder("apple.com");
+        applySignInConfig(call, provider);
+        Task<AuthResult> pendingResultTask = pluginImplementation.getFirebaseAuthInstance().getPendingAuthResult();
+        if (pendingResultTask == null) {
+            nonce = generateNonce(10);
```

Comment:
```suggestion
            nonce = generateNonce(32);
```

In the iOS implementation a length of 32 is used. I have taken this from the [Firebase documentation](https://firebase.google.com/docs/auth/ios/apple).
For this reason, I would also like to increase the length here to `32`.